### DISCUSSION TITLE
How to priorities dependent requests execution order ?

### DISCUSSION THREAD

**Discussion by rabin999** [pact-foundation/nestjs-pact#69] [Q&A]
we have 2 microservices.

**About the Issue**
- multiple pact tests which are dependent on each other.
- verifier need to execute those requests based on their priority

**Example**
1. request A -> Create order with quantity 1 with SKU: `abc`
2. request C -> retrieve list of remaining qty for SKU: `abc`
3. request B -> Create another order with quantity 1 with SKU: `abc`

Execution Order of those requests
- 1 -> 3 -> 2

**Comment by rabin999** [comment]
**Same question in a different repo**

https://github.com/pact-foundation/pact-js/discussions/1174

**Comment by YOU54F** [comment]
this is a code smell, and isn't solely a Pact issue. If you have tests that depend on the outcome of other tests you are going to have a hard time testing

>multiple pact tests which are dependent on each other.

Utilise provider states, to setup the correct state for the provider, for testing each individual scenario.

https://docs.pact.io/provider/using_provider_states_effectively

eg create a product and no product exists, create a product and product exists, try to create a product which already exists. The state will be the world in your Arrange part of the triple A's in test (Arrange, Act, Assert)

**Reply by gorabin** [reply]
@YOU54F 
What is the best way to await to certain time before we verify any tests in provider side ?
For example I would like to add `sleep` time before certain tests.
